src/hash.cr

```diff
@@ -654,14 +665,47 @@ class Hash(K, V)
     @first.try &.value
   end
 
+  # Deletes and returns the first key-value pair in the hash,
+  # or raises `IndexError` if the hash is empty.
+  #
+  # ```
+  # hash = {"foo" => "bar", "baz" => "qux"}
+  # hash.shift
+  # hash # => {"baz" => "qux"}
+  # 
+  # hash = {} of String => String
+  # hash.shift # => Index out of bounds (IndexError)
+  # ```
   def shift
     shift { raise IndexError.new }
   end
 
+  # Same as `#shift`, but returns nil if the hash is empty.
+  #
+  # ```
+  # hash = {"foo" => "bar", "baz" => "qux"}
+  # hash.shift?
+  # hash # => {"baz" => "qux"}
+  # 
+  # hash = {} of String => String
+  # hash.shift # => Index out of bounds (IndexError)
+  # ```
   def shift?
     shift { nil }
   end
 
+  # Deletes and returns the first key-value pair in the hash.
+  # Yields the given block if the hash is empty.
```

Comment:
`Yields to`, yielding is the operation happening on the block, not another operation that the block is passed to.